src/imgbased/plugins/service.py

```diff
@@ -56,8 +56,10 @@ def _generate_iqn(self):
             return
         log.debug("Description=Generate a random iSCSI initiator IQN name")
         suuid = str(uuid.uuid4()).split("-")[-1]
-        factory_f = File("/usr/share/factory/etc/iscsi/initiatorname.iscsi")
-        iqn = factory_f.contents.split(":")[0] + ":" + suuid + "\n"
+        identifier = command.call("/usr/sbin/iscsi-iname")
+        iqn = "InitiatorName="\
+              + identifier.contents.split(":")[0]\
```

Comment:
identifier in this case is the output from the call function, which has no contents attribute, since it's of type 'bytes', not instance of class File.

Reply:
Please also don't forget that you need to decode bytes into str(ing) with i.e. decode('utf-8').

Thus IMHO the new code should look something like i.e. :

identifier = command.call("/usr/sbin/iscsi-iname").decode('utf-8')
iqn = "InitiatorName=" + identifier + "\n"

Reply:
Also please don't forget to remove the line with the suuid assignment, as it's no longer needed in this case.

Reply:
Thanks. Missed those indeed :(
Is there a proper way to (easily) test the code?